src/libifconfig.c

```diff
@@ -401,6 +401,36 @@ ifconfig_get_capability(ifconfig_handle_t *h, const char *name,
 	return (0);
 }
 
+int ifconfig_get_groups(ifconfig_handle_t *h, const char *name,
+    struct ifgroupreq *ifgr)
+{
+	int s, len;
+
+	if (ifconfig_socket(h, AF_LOCAL, &s) != 0) {
+		return (-1);
+	}
+
+	memset(ifgr, 0, sizeof(*ifgr));
+	strlcpy(ifgr->ifgr_name, name, IFNAMSIZ);
+
+	if (ioctl(s, SIOCGIFGROUP, (caddr_t)ifgr) == -1) {
+		if (errno == EINVAL || errno == ENOTTY)
```

Comment:
I'm a little confused as to what this check is for. I know it's in ifconfig already, but I can't find any obvious way for SIOCGIFGROUP to return ENOTTY, and I don't see why EINVAL wouldn't be an error.